jerry-core/ecma/operations/ecma-typedarray-object.c

```diff
@@ -743,68 +915,41 @@ ecma_op_typedarray_from (ecma_value_t items_val, /**< the source array-like obje
   ecma_object_t *new_typedarray_p = ecma_get_object_from_value (new_typedarray);
   ecma_typedarray_info_t info = ecma_typedarray_get_info (new_typedarray_p);
   ecma_typedarray_setter_fn_t setter_cb = ecma_get_typedarray_setter_fn (info.id);
-  ecma_number_t num_var;
 
   /* 17 */
   for (uint32_t index = 0; index < len; index++)
   {
-    /* 17.b */
     ecma_value_t current_value = ecma_op_object_find_by_uint32_index (arraylike_object_p, index);
 
     if (ECMA_IS_VALUE_ERROR (current_value))
     {
-      goto cleanup;
+      ecma_deref_object (new_typedarray_p);
+      ecma_deref_object (arraylike_object_p);
+      return current_value;
     }
 
     if (ecma_is_value_found (current_value))
     {
-      ecma_value_t mapped_value;
-      if (func_object_p != NULL)
+      ecma_value_t error_val = ecma_op_typedarray_from_helper (this_val,
+                                                               current_value,
+                                                               index,
+                                                               func_object_p,
+                                                               info,
+                                                               setter_cb);
+
+      if (ECMA_IS_VALUE_ERROR (error_val))
```

Comment:
This can be ORed (`||`) tho the condition above, and you can always return `ECMA_VALUE_ERROR`.